toolkit/tools/internal/customizationmacros/customizationmacros.go

```diff
@@ -0,0 +1,133 @@
+		"# prevent any locale files from being installed.",
+		"# To enable locale files, remove this file, or comment out '%%_install_langs <LOCALE STRING>'",
+		"# Any packages which are already installed must be reinstalled for this change to take effect.",
+	}
+)
+
+// AddCustomizationMacros adds the currently defined image custimization macros to the specified root directory.
+// For each of disableDocs and DisableRpmLocales a macro file is created with the corresponding macros defined in the
+// default rpm macros directory.
+func AddCustomizationMacros(rootDir string, disableDocs, disableRpmLocales bool, overrideRpmLocales string) (err error) {
+	if overrideRpmLocales != "" && disableRpmLocales {
+		return fmt.Errorf("cannot set both DisableRpmLocales and OverrideRpmLocales")
+	}
+
+	if disableDocs {
+		logger.Log.Debugf("Disabling documentation packages")
+		err = AddMacroFile(rootDir, rpm.DisableDocumentationDefines(nil), DisableDocsMacroFile, docComments)
+		if err != nil {
+			return fmt.Errorf("failed to add disable docs macro file:\n%w", err)
+		}
+	}
+	if disableRpmLocales {
+		logger.Log.Debugf("Disabling locale packages")
+		err = AddMacroFile(rootDir, rpm.DisableLocaleDefines(nil), ConfigureRpmLocalesMacroFile, localeComments)
+		if err != nil {
+			return fmt.Errorf("failed to add disable locales macro file:\n%w", err)
+		}
+	}
+	if overrideRpmLocales != "" {
+		logger.Log.Debugf("Overriding locale packages with (%s)", overrideRpmLocales)
+		err = AddMacroFile(rootDir, rpm.OverrideLocaleDefines(nil, overrideRpmLocales), ConfigureRpmLocalesMacroFile, localeComments)
+		if err != nil {
+			return fmt.Errorf("failed to add override locales macro file:\n%w", err)
+		}
+	}
+	return nil
+}
+
+// validateComments checks that the comments are valid for a macro file: ie they are empty or start with '#'
+func validateComments(comments []string) error {
+	for _, comment := range comments {
+		comment = strings.TrimSpace(comment)
+		if !strings.HasPrefix(comment, "#") && comment != "" {
+			return fmt.Errorf("extra comments must start with '#'(%s)", comment)
+		}
+	}
+	return nil
+}
+
+// AddMacroFile adds a macro file to the specified root directory with the specified macros. The macro file
+// is created in the default rpm macros directory. The macro file will include a default header, with additional comments
+// if desired. Each extra comment should start with a '#' character.
+func AddMacroFile(rootDir string, macros map[string]string, macroFileName string, extraComments []string) error {
+	const macrosPath = "/usr/lib/rpm/macros.d"
+
+	if len(macros) == 0 {
+		return nil
+	}
+
+	header := customizationMacroHeaderComments
+	if len(extraComments) > 0 {
+		extraComments = append(extraComments, "")
+		header = append(customizationMacroHeaderComments, extraComments...)
+	}
+
+	err := validateComments(extraComments)
+	if err != nil {
+		return fmt.Errorf("failed to validate extra comments while adding macro file:\n%w", err)
+	}
```

Comment:
I suggest we drop `validateComments` and just add "#" at the beginning of each line ourselves here. It'll be easier to write the comments and harder to make mistakes.

Reply:
Updated that function to instead ensure the # is present.